src/server/routes/contract/events/paginateEventLogs.ts

```diff
@@ -0,0 +1,193 @@
+    status: Type.String(),
+  }),
+});
+
+responseSchema.example = {
+  result: {
+    cursor: "abcd-xyz",
+    logs: [
+      {
+        chainId: 1,
+        contractAddress: "0x....",
+        blockNumber: 1,
+        transactionHash: "0x...",
+        topics: ["0x..."],
+        data: "0x...",
+        eventName: "TransferFrom",
+        decodedLog: {
+          from: {
+            type: "address",
+            value: "0x...",
+          },
+          to: {
+            type: "address",
+            value: "0x...",
+          },
+          value: {
+            type: "uint256",
+            value: "1000",
+          },
+        },
+        timestamp: 100,
+        transactionIndex: 1,
+        logIndex: 1,
+      },
+    ],
+    status: "success",
+  },
+};
+
+const CursorSchema = z.object({
+  createdAt: z.number().transform((s) => new Date(s)),
+  chainId: z.number(),
+  blockNumber: z.number(),
+  transactionIndex: z.number(),
+  logIndex: z.number(),
+});
+
+export async function pageEventLogs(fastify: FastifyInstance) {
+  fastify.route<{
+    Reply: Static<typeof responseSchema>;
+    Querystring: Static<typeof requestQuerySchema>;
+  }>({
+    method: "GET",
+    url: "/contract/events/paginate-logs",
+    schema: {
+      summary: "Get contract paginated event logs for subscribed contract",
+      description: "Get contract paginated event logs for subscribed contract",
+      tags: ["Contract-Events"],
+      operationId: "pageEventLogs",
+      querystring: requestQuerySchema,
+      response: {
+        ...standardResponseSchema,
+        [StatusCodes.OK]: responseSchema,
+      },
+    },
+    handler: async (request, reply) => {
+      const { cursor, pageSize, topics, contractAddresses } = request.query;
+
+      const standardizedContractAddresses = contractAddresses?.map((val) =>
+        val.toLowerCase(),
+      );
+
+      // add lag behind to account for clock skew, concurrent writes, etc
+      const config = await getConfiguration();
+      const maxCreatedAt = new Date(
+        Date.now() - config.cursorDelaySeconds * 1000,
+      );
+
+      let cursorObj;
+      try {
+        if (cursor) {
+          const decodedCursor = base64.decode(cursor);
+          const parsedCursor = decodedCursor
+            .split("-")
+            .map((val) => parseInt(val));
+          const [createdAt, chainId, blockNumber, transactionIndex, logIndex] =
+            parsedCursor;
+          const validationResult = CursorSchema.safeParse({
+            createdAt,
+            chainId,
+            blockNumber,
+            transactionIndex,
+            logIndex,
+          });
+
+          if (!validationResult.success) {
+            throw new Error("Invalid cursor format");
+          }
+
+          cursorObj = validationResult.data;
+        }
+      } catch (error) {
+        throw createCustomError(
+          "Invalid cursor",
+          StatusCodes.BAD_REQUEST,
+          "BAD_REQUEST",
+        );
+      }
```

Comment:
can we put this logic inside getTransactionReceiptsByCursor() so that others using this function can use the logic